packages/splits-v2/src/splitters/SplitFactoryV2.sol

```diff
@@ -48,14 +51,23 @@ abstract contract SplitFactoryV2 is Nonces {
         external
         returns (address split)
     {
-        split = Clone.cloneDeterministic({
+        bytes32 salt = _getSalt({ _splitParams: _splitParams, _owner: _owner, _salt: _salt });
+
+        split = Clone.predictDeterministicAddress({
             _implementation: SPLIT_WALLET_IMPLEMENTATION,
-            _salt: _getSalt({ _splitParams: _splitParams, _owner: _owner, _salt: _salt })
+            _salt: salt,
+            _deployer: address(this)
         });
 
+        if (split.code.length > 0) {
+            return split;
+        }
+
+        split = Clone.cloneDeterministic({ _implementation: SPLIT_WALLET_IMPLEMENTATION, _salt: salt });
```

Comment:
hm is it more performant to make the clone in a try/catch and then do the predict / code length check in the catch? makes the happy-case better (which should be ~100% of usage?)

don't feel strongly about it.. not sure how much we're actually shaving off